tests/test_redirection.rs

```diff
@@ -4,12 +4,16 @@ use std::time::{Duration, Instant};
 
 use attohttpc::ErrorKind;
 use lazy_static::lazy_static;
-use rouille::Response;
+use rouille::{router, Response};
 
 lazy_static! {
     static ref STARTED: bool = {
         thread::spawn(move || {
-            rouille::start_server("localhost:55123", move |_| Response::redirect_301("/"));
+            rouille::start_server("localhost:55123", move |request| router!(request,
```

Comment:
I wonder if using `rouille` is really necessary here. Using a fixed response should be sufficient and binding on a free random instead of fixed high port seems advisable, c.f. `test_timeout.rs`. This is the only place we are using this and lLoosing `rouille` would lighten the development dependencies significantly which would be nice IMHO.

Reply:
I personally don't mind depending on rouille for development only. It's important for me to have end to end tests. I also think it's easier to write them this way.

I agree on the port thing though, it would be preferable to use a random unallocated port.

Reply:
> It's important for me to have end to end tests. 

I totally agree. But we can have those and fix the port thing by dropping `rouille` and add single call to `stream.write_all(b"some hard-coded HTTP response").unwrap();` into the `test_timeout.rs` example.

Reply:
We could also avoid the `// Wait until server is ready. 10s timeout in case of error creating server.` part as we could bind the socket on the main thread only put the accept loop onto a background thread.

Reply:
Create an example commit on how we could do that: https://github.com/adamreichold/attohttpc/commit/580abf0d66c17b226ec8214b4b0a91cd64dd1e1b